apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/KmsTest.cs

```diff
@@ -0,0 +1,247 @@
+﻿// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Apis.Storage.v1.Data;
+using System;
+using System.IO;
+using Xunit;
+using static Google.Apis.Storage.v1.Data.Bucket;
+using Object = Google.Apis.Storage.v1.Data.Object;
+
+namespace Google.Cloud.Storage.V1.IntegrationTests
+{
+    /// <summary>
+    /// Tests for customer-managed encryption keys (CMEK) that use Cloud KMS.
+    /// </summary>
+    [Collection(nameof(StorageFixture))]
+    public class KmsTest
+    {
+        private static string KeyName1 { get; } = Environment.GetEnvironmentVariable("KMS_TEST_KEY_1");
+        private static string KeyName2 { get; } = Environment.GetEnvironmentVariable("KMS_TEST_KEY_2");
+
+        private static string VersionedKeyName1Prefix => KeyName1 + "/cryptoKeyVersions/";
+        private static string VersionedKeyName2Prefix => KeyName2 + "/cryptoKeyVersions/";
+
+        private static bool SkipTests => string.IsNullOrEmpty(KeyName1) || string.IsNullOrEmpty(KeyName2);
+
+        private readonly StorageFixture _fixture;
+
+        public KmsTest(StorageFixture fixture) => _fixture = fixture;
+
+        [SkippableFact]
+        public void UploadObject_BucketHasDefaultKmsKey_UploadWithNoSpecifiedKey()
+        {
+            Skip.If(SkipTests);
```

Comment:
Will CI test fail if tests are skipped? I think this would be the behaviour we want.

Reply:
No - and in fact CI won't use this yet as it doesn't have those environment variables. I don't know whether it's on the right whitelist, either. I'll have a think about this...